docs/release_process.md

```diff
@@ -278,7 +283,7 @@ git push --tags
 - Publish the release.
 
 > **_NOTE:_** In case there is a need to run the job again, delete the release draft and run the
-GitHb Action again.
+Github Action again.
 
 Using `release_helper.py` script:
```

Comment:
Doesn't using `release_helper.py` duplicate some of the commands above (making a tag and pushing it)?